File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) { 
        int left = 0;
        int right = heights.length - 1;
        int maxArea = 0;

        while(left < right)
        {
            int x = right - left;
            int y = Math.min(heights[right], heights[left]);
            maxArea = Math.max(maxArea, x*y);
            if(heights[right] < heights[left])
            {
                right--;
            }
            else
            {
                left++;
            }

        }
        return maxArea;

        
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-2.py
class Solution:
    def trap(self, height: List[int]) -> int:
        # find leftmax, right max
        # but min of those take and calculate for each i
        l, r = 0, len(height) - 1
        leftMax = height[l]
        rightMax = height[r]
        res = 0

        while l < r:
            if leftMax < rightMax:
                l += 1 # this is before anything not in the end
                leftMax = max(leftMax, height[l])
                res += leftMax - height[l]
                #l += 1 NOT HERE
            else:
                r -= 1
                rightMax = max(rightMax, height[r])
                res += rightMax - height[r]
        
        return res

        
        